### PR TITLE
Issue 12537 - Templatizing opEquals results in infinite recursion in the compiler

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=12537

### CODE REVIEW

File: src/dtemplate.d [hunk 1]
```diff
@@ -1531,14 +1534,15 @@ public:
                     if (m == MATCHnomatch)
                     {
                         AggregateDeclaration ad = isAggregate(farg.type);
-                        if (ad && ad.aliasthis)
+                        if (ad && ad.aliasthis && argtype0 != att)
                         {
+                            if (!att && argtype0.checkAliasThisRec())   // Bugzilla 12537
+                                att = argtype0;
```
Comment (WalterBright) [inline src/dtemplate.d]: This works with two mutually referring aliases, but what if there are three or more in the cycle?

Comment (9rnsr) [inline src/dtemplate.d]: Ah, my fault. Fixed.

**Comment by 9rnsr** [comment]
@WalterBright This is ready to merge.

**Comment by ibuclaw** [comment]
Looks reasonable to me.  @WalterBright ?

**Comment by 9rnsr** [comment]
@WalterBright Ping

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**